src/libcpu/src/jit/jit.cpp

```diff
@@ -65,19 +74,39 @@ initStubs()
    auto introLabel = a.newLabel();
    auto extroLabel = a.newLabel();
    auto exitLabel = a.newLabel();
+   auto verifyPreLabel = a.newLabel();
+   auto verifyPostLabel = a.newLabel();
 
    // This is invoked to set up the neccessary state for
    //  the JIT block we want to execute.
    a.bind(introLabel);
    a.push(asmjit::x86::rbp);
    a.push(asmjit::x86::rbx);
-   a.push(asmjit::x86::rdi);
+#ifdef PLATFORM_WINDOWS
+   a.push(asmjit::x86::rdi);  // RSI/RDI are callee-saved in Windows x64 ABI
    a.push(asmjit::x86::rsi);
+#endif
    a.push(asmjit::x86::r12);
    a.push(asmjit::x86::r13);
    a.push(asmjit::x86::r14);
    a.push(asmjit::x86::r15);
-   a.sub(asmjit::x86::rsp, 0x38);
+   if (gJitMode == jit_mode::verify) {
+      // Leave space for the verification buffer on the stack:
+      //  RSP+0 = Windows argument shadow space (reserved on all platforms for simplicity)
+      //  RSP+32 = size of this stack area (for reference on return)
+      //  RSP+36 = PPC instruction being verified
+      //  RSP+40 = verification buffer
+      uint32_t stackSpace = 40 + static_cast<uint32_t>(sizeof(VerifyBuffer));
+      // Preserve 16-byte stack alignment -- note that RSP comes in unaligned from the CALL instruction
+      stackSpace = align_up(stackSpace, 8);
+      if ((stackSpace & 15) == 0) {
+         stackSpace += 8;
+      }
+      a.sub(asmjit::x86::rsp, stackSpace);
```

Comment:
This is very dangerous, the stack unwind meta-data generator relies on constants for this stuff.  We might need to make it so the stack size selected is passed in or something.